vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_source.go

```diff
@@ -0,0 +1,121 @@
+package metrics
+
+import (
+	"context"
+	k8sapiv1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/model"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	externalmetricsv1beta1 "k8s.io/metrics/pkg/apis/external_metrics/v1beta1"
+	"k8s.io/metrics/pkg/apis/metrics/v1beta1"
+	"time"
+
+	resourceclient "k8s.io/metrics/pkg/client/clientset/versioned/typed/metrics/v1beta1"
+	"k8s.io/metrics/pkg/client/external_metrics"
+)
+
+type Source interface {
```

Comment:
I think we should not introduce a new `Source` interface and try to keep the existing `PodMetricsesGetter`, even if that means an unimplemented/empty `Watch` method for now

Reply:
It can be done, but it's a messier setup, and not just in saving lines of code.   The name `Source` does suck, so if you have any suggestions for that, I'm all ears.

There are two issues with it:
1. The `PodMetricsGetter` and `PodMetricsInterface` interfaces aren't part of the VPA, they're part of the Metrics API (the VPA just has auto-gen'd client interface) from the `metrics-server`.  While the External Metrics API is a separate peer system, hiding it behind a peer's interface creates an unnecessary coupling of these two separate systems as we're using them here.  An abstraction over both systems (Metrics and External Metric) is more ontologically sound.  We don't want the External Metrics system here to need updates whenever the Metrics system changes.
2.  The caller first gets a `PodMetricsGetter`, calls its `PodMetrices(namespace)` method, gets the returned `PodMetricsInterface`, and then calls that objects' `List()` method.  So we'd have to fake two interfaces, and we'd end up with more code for a less maintainable setup.

Reply:
@dbenque what do you think ?

Reply:
I did a first review... It is a bit late for me now, I will try to look at that particular point ASAP.